spec/component_reader.spec.js

```diff
@@ -1,138 +1,129 @@
-        });
-    });
-
-    it('Should return appropriate error if trigger file is missing', () => {
-
-        var reader = new ComponentReader();
-
-        var promise = reader.init('/spec/component/')
-            .then(() => reader.loadTriggerOrAction('missing_trigger'));
+/* eslint-disable no-unused-expressions */
+const { expect } = require('chai');
+const { ComponentReader } = require('../lib/component_reader.js');
 
-        waitsFor(() => promise.isFulfilled() || promise.isRejected(), 10000);
+const waitForThen = (fn, testFn) => {
+  fn;
+  testFn;
+};
 
-        runs(() => {
-            expect(promise.isRejected()).toEqual(true);
-            var err = promise.inspect().reason;
-            expect(err.message).toMatch(
-                //eslint-disable-next-line no-useless-escape
-                /Failed to load file \'.\/triggers\/missing_trigger.js\': Cannot find module.+missing_trigger\.js/
-            );
-            expect(err.code).toEqual('MODULE_NOT_FOUND');
-        });
-    });
-
-    it('Should return appropriate error if missing dependency is required by module', () => {
-
-        var reader = new ComponentReader();
-
-        var promise = reader.init('/spec/component/')
-            .then(() => reader.loadTriggerOrAction('trigger_with_wrong_dependency'));
-
-        waitsFor(() => promise.isFulfilled() || promise.isRejected(), 10000);
-
-        runs(() => {
-            expect(promise.isRejected()).toEqual(true);
-            var err = promise.inspect().reason;
-            expect(err.message).toEqual(
-                'Failed to load file \'./triggers/trigger_with_wrong_dependency.js\': '
-                + 'Cannot find module \'../not-found-dependency\''
-            );
-            expect(err.code).toEqual('MODULE_NOT_FOUND');
-        });
-    });
-
-    it('Should return appropriate error if trigger file is presented, but contains syntax error', () => {
-
-        var reader = new ComponentReader();
-
-        var promise = reader.init('/spec/component/')
-            .then(() => reader.loadTriggerOrAction('syntax_error_trigger'));
-
-        waitsFor(() => promise.isFulfilled() || promise.isRejected(), 10000);
-
-        runs(() => {
-            expect(promise.isRejected()).toEqual(true);
-            var err = promise.inspect().reason;
-            expect(err.message).toEqual(
-                "Trigger or action 'syntax_error_trigger' is found, but can not be loaded. "
-                + "Please check if the file './triggers/syntax_error_trigger.js' is correct."
-            );
-        });
+describe('Component reader', () => {
+  it('Should find component located on the path', () => {
+    const reader = new ComponentReader();
+    const promise = reader.init('/spec/component/');
+
+    waitForThen(() => promise.isFulfilled() || promise.isRejected(),
+      () => {
+        expect(promise.isFulfilled()).to.be.true;
+        expect(reader.componentJson.title).to.be.equal('Client component');
```

Comment:
```suggestion
        expect(reader.componentJson.title).to.equal('Client component');
```

nitpick for future: you can drop the `.be`, it's just a language chainer

Reply:
👍